File: refrence-service/src/main/java/com/kapitonau/ps/refrenceservice/controller/ReferenceItemController.java
package com.kapitonau.ps.refrenceservice.controller;

import com.kapitonau.ps.apirequestlib.reference.api.ReferenceItemApi;
import com.kapitonau.ps.apirequestlib.reference.dto.ReferenceItemResponse;
import com.kapitonau.ps.refrenceservice.service.ReferenceItemService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequiredArgsConstructor
public class ReferenceItemController implements ReferenceItemApi {

    private final ReferenceItemService referenceItemService;

    @Override
    public List<ReferenceItemResponse> getAllReferenceItem(Long spaceId, String lang, String referenceType, String itemCode) {
        return referenceItemService.getAllReferenceItem(spaceId, lang, referenceType, itemCode);
    }

    @Override
    public ReferenceItemResponse getReferenceItem(Long spaceId, String lang, Long referenceItemId) {
        return referenceItemService.getReferenceItem(spaceId, lang, referenceItemId);
    }
}

File: reference-service/src/main/kotlin/com/kapitonau/referenceservice/controller/ReferenceItemController.kt
package com.kapitonau.referenceservice.controller

import com.kapitonau.projectstudio.referenceservice.api.ReferenceItemApi
import com.kapitonau.projectstudio.referenceservice.dto.ReferenceItemResponse
import com.kapitonau.referenceservice.service.ReferenceItemService
import org.springframework.web.bind.annotation.RestController

@RestController
class ReferenceItemController(
    private val referenceItemService: ReferenceItemService
) : ReferenceItemApi {
    override fun getReferenceItemById(referenceItemId: Long): ReferenceItemResponse =
        referenceItemService.geReferenceItemById(referenceItemId)

    override fun getAllReferenceItems(
        referenceType: String,
        itemCode: String?
    ): List<ReferenceItemResponse?>? =
        referenceItemService.getReferenceByReferenceTypeAndItemCode(referenceType, itemCode)
}